### ISSUE TITLE
Hookshot doesn't verify it's sessions

### ISSUE THREAD

**Issue by Twi1ightSparkle** [matrix-org/matrix-hookshot#1001]
This generates red warning on all it's messages

![screenshot_2024-12-11T14-19-15Z](https://github.com/user-attachments/assets/09150c6c-6f25-4bda-be5a-12e806497ab1)

![screenshot_2024-12-11T14-19-41Z](https://github.com/user-attachments/assets/a20683f2-aed8-4539-898a-168f15535aba)

And E2EE Hookshot will stop working entirely once [MSC4153: Exclude non-cross-signed devices](https://github.com/matrix-org/matrix-spec-proposals/pull/4153) gets implemented.

**Comment by Half-Shot** [comment]
This requires https://github.com/element-hq/matrix-bot-sdk/, and the rust sdk bindings to support cross signing. This work isn't trivial, so I'll ask about it.

**Comment by szymczag** [comment]
Thanks, @Twi1ightSparkle, for bringing that up — I was just about to create a new issue for this.

Hello, @Half-Shot, is there any chance you could get this fixed in the near future?

**Comment by Half-Shot** [comment]
Hey hey, this is currently being looked at in https://github.com/element-hq/element-meta/issues/2709

**Comment by szymczag** [comment]
@Half-Shot thanks! I can see that the work is in progress, so I'll keep my fingers crossed and wait patiently ;)

**Comment by Half-Shot** [comment]
I aim to not keep your fingers crossed for too long then :)

**Comment by szymczag** [comment]
> I aim to not keep your fingers crossed for too long then :)

Hi @Half-Shot, any news? :)

**Comment by Half-Shot** [comment]
Hey, yes on this front. The Crypto team will be extending https://github.com/matrix-org/matrix-rust-sdk-crypto-nodejs/pulls to support verification, which then means hookshot will make use of it. https://github.com/element-hq/element-meta/issues/2751 will also land so that at least manual verification will become possible.

I don't know the timelines for this work yet, but it must happen before https://github.com/element-hq/element-meta/issues/2700 happens.

**Comment by andybalaam** [comment]
**Workaround**

https://github.com/element-hq/element-meta/issues/2751 is now done, meaning that if you log in to EW as the bot's user and you have the ed25519 fingerprint ("security key") of the device the bot is using, you can manually tell EW to sign the device, making the
bot verified.

Instructions on how to do this (on the EW side) are at: https://github.com/element-hq/element-web/pull/30094

I'm not sure how easy it is to get the fingerprint out of Hookshot. Hopefully @Half-Shot knows

**Comment by Tony763** [comment]
> **Workaround**
> 
> [element-hq/element-meta#2751](https://github.com/element-hq/element-meta/issues/2751) is now done, meaning that if you log in to EW as the bot's user and you have the ed25519 fingerprint ("security key") of the device the bot is using, you can manually tell EW to sign the device, making the bot verified.
> 
> Instructions on how to do this (on the EW side) are at: [element-hq/element-web#30094](https://github.com/element-hq/element-web/pull/30094)
> 
> I'm not sure how easy it is to get the fingerprint out of Hookshot. Hopefully [@Half-Shot](https://github.com/Half-Shot) knows

Hello, does this workaround still work? I logged in as hookshot user, but rooms nor direct messages wont load. Sessions show, that it cannot be verified. (Encrypted message from hookshot works, messages to it not - ends with No users in room, cannot decrypt)

<img width="492" height="263" alt="Image" src="https://github.com/user-attachments/assets/92131d0c-f290-4638-bda5-40d2c0bdb6eb" />

**Comment by andybalaam** [comment]
Yes, this workaround does work. You need to follow the instructions at https://github.com/element-hq/element-web/pull/30094
